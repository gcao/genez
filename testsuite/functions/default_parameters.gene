(fn greet [name = "World"] 
    (println #"Hello, #{name}!"))

(greet)
(greet "Gene")

(fn add [x y = 10]
    (println (+ x y)))

(add 5)
(add 5 3)

(fn connect [host = "localhost" port = 8080]
    (println #"#{host}:#{port}"))

(connect)
(connect "example.com")
(connect "example.com" 443)

(fn format [content prefix = "[" suffix = "]"]
    (println #"#{prefix}#{content}#{suffix}"))

(format "INFO")
(format "ERROR" "{")
(format "DEBUG" "<" ">")