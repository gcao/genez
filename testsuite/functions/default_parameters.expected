Hello, World!
Hello, Gene!
15
8
localhost:8080
example.com:8080
example.com:443
[INFO]
{ERROR]
<DEBUG>